docs/signing-spec.md

```diff
@@ -342,6 +343,236 @@ For example, if a package is signed by an unknown host key, then this package ha
+
+Consequently, when a bindle is yanked, the invoice MUST be updated in two ways:
+
+- The top-level `yanked` field MUST be set to true
+- The host key MUST generate a `yanked_signature` composed by signing the following block of data:
+
+```
+Matt Butcher <matt.butcher@example.com>
+mybindle
+0.1.0
+host
+1611960337
+yanked
+~
+A security flaw was found. See CVE-1234 for details.
+```
+
+This differs from the regular signature block in four important ways:
+- The only role that is allowed to sign in this way is the `host` role
+- The timestamp is the time at which the host marked this bindle as yanked
+- The word `yanked` appears below the timestamp. This is a trivial guard against an attack that attempts to repurpose a parcel-less signature as a yank block.
+- After the `~` divider, the optional `yanked_reason` is included. If a `yanked_reason` is provided on the invoice, it MUST be included in the signature block to prevent tampering.
+
+To verify a `yanked_signature`, an agent:
+
+- MUST verify that the host key used to sign is in the agent's keyring
+    - An agent MAY refuse to treat a bindle as yanked if the key is not known
+    - An agent SHOULD notify the user if a bindle is yanked, but the yanking host key is not known
+- MUST verify that the signature is valid (using the same formula described for regular signature validation).
+- SHOULD verify that the timestamp for yanking is not the same as the signature block's `at` timestamp.
+    - This is a trivial protection against attempts to forge a yank block.
+- SHOULD require that the host key used to yank is also the same host key originally used to sign
+    - This is complicated by host key rotation. Our recommendation is that on host key rotation, all packages should be re-signed using the new host key so that it is always clear who the host authority is.
+
+
+The format of the `yanked_signature` is the same as the format of a regular signature:
+
+```toml
+bindle_version = "1.0.0"
+yanked = true
+
+[[yanked_signature]]
+by = "Matt Butcher <matt.butcher@example.com>"
+signature = "ddd237895ac..."
+key = "1c44..."
+role = "host"
+at = 1611960337
+```
+
+Note that this strategy does not prevent a key from being used to generate a known-bad signature so that the validation fails.
+
+### Known Issues with Yank Signing
+
+The current model does not provide a way to say "this package should never be trusted again" except for generating known-bad signatures.
+
+Likewise, while signing with `host` keys indicates that the host has yanked the package, there is no clear way to indicate the creator's intent, or a verifier's stance. The reason for this is that yanking is a host-specific operation, and does not have
+to do directly with creator intent. (In other words, the creator is not the authority on whether a bindle is yanked. The host is.)
+
+A compromised host could yank all packages, which may have dire initial consequences.
```

Comment:
Is there the possibility of a compromised host un-yanking packages?

Reply:
Oh! That is a very good point. That could be bad.

Reply:
Okay, for now I have noted the problem, but I don't have a solution.